### PR TITLE
Refactor test scripts in preparation for CI migration

### PR DESCRIPTION
**Pull request by chewong**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind test

**What this PR does / why we need it**:
- Made bash scripts cleaner and more consistent
- Automate the creation of `$AZURE_CREDENTIAL_FILE` in `test/credentials` instead of using `run-tests-all-clouds.sh`. It will either grab environment variables (`$tenantId`, `$subscriptionId`, `$aadClientId`, and `$aadClientSecret`) to build `$AZURE_CREDENTIAL_FILE` if you are testing locally, or if it is being tested on Prow, it will grab [`$AZURE_CREDENTIALS`](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/cloud-provider-azure/cloud-provider-azure-config.yaml#L5) to build `$AZURE_CREDENTIAL_FILE`
- Automate the creation and deletion of test resource groups under Azure subscription described in `$AZURE_CREDENTIAL_FILE` for sanity and integration tests
- [Link to Prow job config](https://github.com/chewong/test-infra/blob/azurefile-csi-driver-config/config/jobs/kubernetes-sigs/azurefile-csi-driver/azurefile-csi-driver-config.yaml)
- For E2E test on Prow, [this](https://github.com/chewong/test-infra/blob/azurefile-csi-driver-config/kubetest/azure.go#L1012-L1018) will get triggered

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```

```

### CODE REVIEW

File: test/integration/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	assert.NoError(t, err)
+
+	ctx := context.Background()
+	// Create an empty resource group for integration test
+	t.Logf("Creating resource group %s in %s", creds.ResourceGroup, creds.Cloud)
+	_, err = azureClient.EnsureResourceGroup(ctx, creds.ResourceGroup, creds.Location, nil)
+	assert.NoError(t, err)
+	defer func() {
+		t.Logf("Deleting resource group %s in %s", creds.ResourceGroup, creds.Cloud)
+		err := azureClient.DeleteResourceGroup(ctx, creds.ResourceGroup)
```
Comment (andyzhangx) [inline test/integration/integration_test.go]: can we keep the resource group? usually in the integration test, if there is a volume creation, then should be a volume deletion, so no need to delete whole resource group.

**Comment by chewong** [comment]
@andyzhangx I am assuming we will move all the tests from Travis CI to prow because they were all included in https://github.com/kubernetes/test-infra/pull/12386

**Comment by andyzhangx** [comment]
> @andyzhangx I am assuming we will move all the tests from Travis CI to prow because they were all included in [kubernetes/test-infra#12386](https://github.com/kubernetes/test-infra/pull/12386)

great! Then how do you deal with the driver image in the test, currently they are under `mcr.microsoft.com/k8s/csi/azurefile-csi`

File: test/integration/integration_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+	// Test on AzureChinaCloud
+	creds, err = credentials.CreateAzureCredentialFile(true)
+	defer func() {
+		assert.NoError(t, credentials.DeleteAzureCredentialFile())
+	}()
+
+	// Skip the test if Azure China cloud credentials are not supplied
+	if err != nil {
+		t.Skip()
```
Comment (ZeroMagic) [inline test/integration/integration_test.go]: I think we need a log about the error here. We have set [environment variables](https://travis-ci.org/kubernetes-sigs/azurefile-csi-driver/jobs/583910637#L189) related to AzureChinaCloud in CI. But the test was skipped. Have a look at the [link](https://travis-ci.org/kubernetes-sigs/azurefile-csi-driver/jobs/583910637#L703).

**Pull request merged**